lib/src/middleware_api.dart

```diff
@@ -0,0 +1,30 @@
+library middleware_api;
+
+import 'package:greencat/src/action.dart';
+import 'package:greencat/src/store.dart';
+import 'package:greencat/src/typedefs.dart';
+
+/// Middleware is the suggested way to extend Redux with custom functionality.
+/// Middleware lets you wrap the store’s dispatch method for fun and profit.
+/// The key feature of middleware is that it is composable. Multiple middleware
+/// can be combined together, where each middleware requires no knowledge of
+/// what comes before or after it in the chain.
```

Comment:
Is this really needed given we already have [StreamTransformer](https://api.dartlang.org/stable/1.20.1/dart-async/StreamTransformer-class.html) in the standard library?

Could the incoming actions and outgoing state objects be arranged into stream pipelines. Some strawman ASCII art:

```
ActionSink > AT1 > AT2 > AT3 > Store > Reducer > ST1 > ST2 > ST3 > Widget/Component
```

Where `AT#` stands for "action transformer" and `ST#` stands for "state transformer" and are implemented as `StreamTransformer`s.